### PR TITLE
Create GitHub Actions for linting, testing and releasing

### PR DESCRIPTION
**Pull request by christianhueserhzdr**
Extend CI pipeline to do the linting and testing on each pull request and on each merging into branch main and tagging releases and to do the releasing on each tag created.

In preparation of the relocation from GitLab.com to GitHub as described in: https://gitlab.com/hifis/ansible/gitlab-role/-/issues/49

### CODE REVIEW

**Comment by christianhueserhzdr** [comment]
Dear Tobias @tobiashuste, I prepared the GitHub Actions to move developments from GitLab to GitHub. Could you please have a look whether all is fine and I did not miss anything? Thanks for your review.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      - "main"
+    tags:
+      - "v*.*.*"
+
+env:
+  PY_COLORS: 1
+  ANSIBLE_FORCE_COLOR: 1
+  WORKON_HOME: ~/.cache/venv
+  PIPENV_CACHE_DIR: ~/.cache/pipenv
+  PIP_CACHE_DIR: ~/.cache/pip
```
Comment (tobiashuste) [inline .github/workflows/ci.yml:21]: I just learned about the caching feature of the `setup-python` action and suggest to use this instead.

```suggestion
```

File: .github/workflows/prepare-action/action.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        
+    - name: Restore Python dependency cache.
+      uses: actions/cache@v3
+      with:
+        path: ~/.cache
+        key: ${{ runner.os }}-python-${{ steps.setup-python.outputs.python-version }}-pipenv-${{ hashFiles('Pipfile.lock') }}
+
+    - name: Install pipenv.
+      run: pip3 install --ignore-installed pipenv
+      shell: bash
```
Comment (tobiashuste) [inline .github/workflows/prepare-action/action.yml:26]:
```suggestion
```

File: .github/workflows/prepare-action/action.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# SPDX-FileCopyrightText: 2022 Helmholtz-Zentrum Dresden-Rossendorf (HZDR)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+---
+name: Install dependencies and prepare the environment.
+description: Install the necessary dependencies for jobs.
+runs:
+  using: composite
+  steps:
```
Comment (tobiashuste) [inline .github/workflows/prepare-action/action.yml:11]:
```suggestion
  steps:
    - name: Install pipenv
      run: pipx install pipenv
      shell: bash

```

File: .github/workflows/prepare-action/action.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    - name: Set up Python 3.
+      uses: actions/setup-python@v3
+      id: setup-python
+      with:
+        python-version: '3.9'
```
Comment (tobiashuste) [inline .github/workflows/prepare-action/action.yml:16]: According to https://github.com/actions/setup-python#caching-packages-dependencies

```suggestion
        python-version: '3.9'
        cache: 'pipenv'
```

**Review by tobiashuste — CHANGES_REQUESTED** [review]
Thanks for implementing this! I found out about the caching feature of the `setup-python` action which is a bit simpler to apply. This is why I suggested changes to make use of it.

Could you please also add the Changelog entry?

**Comment by tobiashuste** [comment]
@christianhueserhzdr Is this ready from your side?

**Comment by christianhueserhzdr** [comment]
> @christianhueserhzdr Is this ready from your side?

Dear Tobias @tobiashuste, from my side I incorporated your suggestions, added the Dependabot to the GitHub Actions and removed the obsolete GitLab CI file. Could you please review again? Thanks.

**Review by tobiashuste — APPROVED** [review]
Thanks, @christianhueserhzdr for implementing this :shipit:

**Pull request merged**